local_planner/src/nodes/local_planner.cpp

```diff
@@ -61,6 +61,9 @@ void LocalPlanner::dynamicReconfigureSetParams(avoidance::LocalPlannerNodeConfig
 void LocalPlanner::setGoal(const Eigen::Vector3f& goal) {
   goal_ = goal;
   ROS_INFO("===== Set Goal ======: [%f, %f, %f].", goal_.x(), goal_.y(), goal_.z());
+  Eigen::Vector2f pos_xy = position_.head<2>();
+  Eigen::Vector2f goal_xy = goal_.head<2>();
+  pos_to_goal_dist_xy_ = (position_ - goal_).head<2>().norm();
```

Comment:
I recall defining a macro for this at some point...
Consider moving it to common so we can use it everywhere?
https://github.com/PX4/avoidance/blob/3acc9a55d7a57e0c34919087637ea50520bf045b/local_planner/src/nodes/waypoint_generator.cpp#L8